packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -128,12 +173,15 @@ export const plugin = <TContext>(
       function didEnd(
         requestContext:
           | GraphQLRequestContextWillSendResponse<TContext>
-          | GraphQLRequestContextDidEncounterErrors<TContext>,
+          | GraphQLRequestContextDidEncounterErrors<TContext>
+          | GraphQLRequestContextDidResolveOperation<TContext>,
       ) {
         if (endDone) return;
         endDone = true;
         treeBuilder.stopTiming();
 
+        if (metrics.captureTraces === false) return;
```

Comment:
Understanding why this is guaranteed to not be undefined is a bit hairy. I see you have some comments about that which is great. It would be nice to move shouldTraceOperation into this function but I see that the requestContext types are a bit weird. (Honestly I hope we can eventually make some improvements to the pipeline API to allow this whole function to just be responseDidEnd but @abernix explained to me that there are a few errors that don't hit responseDidEnd, sigh.)

Can you at least assert that metrics.captureTraces is defined?

Reply:
I added a log warn statement.